vllm_v1/request.py

```diff
@@ -0,0 +1,121 @@
+import enum
+from dataclasses import dataclass
+from typing import (TYPE_CHECKING, Any, Callable, Dict, List, Mapping,
+                    Optional, Union)
+
+from vllm.sampling_params import SamplingParams
+from vllm.lora.request import LoRARequest
+
+if TYPE_CHECKING:
+    from vllm.inputs import LLMInputs
+    from vllm.multimodal.base import MultiModalDataDict
+
+
+class Request:
+
+    def __init__(
+        self,
+        request_id: str,
+        inputs: "LLMInputs",
+        arrival_time: float,
+        sampling_params: SamplingParams,
+        lora_request: Optional[LoRARequest] = None,
+    ) -> None:
+        self.request_id = request_id
+        self.inputs = inputs
+        self.metrics = RequestMetrics(arrival_time=arrival_time,
+                                      last_token_time=arrival_time,
+                                      first_scheduled_time=None,
+                                      first_token_time=None,
+                                      time_in_queue=None)
+        self.sampling_params = sampling_params
+        self.lora_request = lora_request
+
+        self.status = RequestStatus.WAITING
+        self.stop_reason: Union[int, str, None] = None
+        assert sampling_params.max_tokens is not None
+        self.max_tokens = sampling_params.max_tokens
+
+        self.prompt = inputs.get("prompt")
+        self.prompt_token_ids = inputs["prompt_token_ids"]
+        self.num_prompt_tokens = len(self.prompt_token_ids)
+        self.output_token_ids: List[int] = []
+        self.output_text = ""
```

Comment:
I think there's no reason to not separate them out?

Besides the things you listed, this is required because `RequestOutput` distinguishes the prompt and output.